### PR TITLE
Refactor NewMonsterAnim and MonsterGraphic enum

### PR DESCRIPTION
**Pull request by obligaron**
After #2354 and #2361 I thought about refactoring monster animation logic a little bit.

Content:

- Less locations where `_mdir` is set
- Less locations that directly work with `AnimStruct`
- Scoped enum `MonsterGraphic`

... and there is much more to refactor but step for step 😉

### CODE REVIEW

File: Source/monster.h [hunk 1]
```diff
@@ -87,13 +87,13 @@ enum MON_MODE : uint8_t {
...
-	MA_GOTHIT,
-	MA_DEATH,
-	MA_SPECIAL,
+enum class MonsterGraphic {
+	Stand,
+	Walk,
+	Attack,
+	GotHit,
+	Death,
+	Special,
```
Comment (ephphatha) [inline Source/monster.h:96]: 😍

File: Source/monster.h [hunk 2]
```diff
@@ -127,6 +127,13 @@ struct CMonster {
 	/** placeflag enum as a flags*/
 	uint8_t mPlaceFlags;
 	AnimStruct Anims[6];
+	/**
+	 * @brief Returns AnimStruct for specified graphic
```
Comment (ephphatha) [inline Source/monster.h:131]: Bit nitpicky but this should document the param and return despite those being pretty obvious from the name.

File: Source/dead.cpp [hunk 3]
```diff
@@ -16,13 +16,14 @@ DeadStruct Dead[MaxDead];
 int8_t stonendx;
 
 namespace {
-void InitDeadAnimationFromMonster(DeadStruct &dead, const CMonster &mon)
+void InitDeadAnimationFromMonster(DeadStruct &dead, CMonster &mon)
```
Comment (ephphatha) [inline Source/dead.cpp:19]: Providing a const version of CMonster::GetAnimData should allow keeping mon as a const ref

**Review by ephphatha — CHANGES_REQUESTED** [review]
Love it, only suggestion I hope can be implemented is adding a const form of GetAnimData to save changing the signature of InitDeadAnimationFromMonster.

File: Source/monster.h (hunk 2 above)
Comment (obligaron) [reply Source/monster.h:131]: I'm not a fan of inserting comments that doesn't help anyone.
If you really want it, I can add it.

Something like this?

```
	/**
	 * @brief Returns AnimStruct for specified graphic
	 * @param graphic Specifies which AnimStruct should be returned
	 * @return AnimStruct for the graphic
	 */
```

File: Source/dead.cpp (hunk 3 above)
Comment (obligaron) [reply Source/dead.cpp]: I got a error in `AnimationInfo` and thought I would need to update many places.
But after trying it, it wasn't many places that hand to be adjusted.
So I added it (last commit) 👍🙂

File: Source/monster.h (hunk 2 above)
Comment (AJenbo) [reply Source/monster.h:131]: I'm fine with leaving out lines that aren't helpful

Comment (ephphatha) [reply Source/monster.h:131]: In that case no need to worry about this :)

**Review by ephphatha — APPROVED** [review]

**Pull request merged**